OWML.ModHelper.Menus/ModComboInput.cs

```diff
@@ -0,0 +1,148 @@
+﻿using OWML.Common.Menus;
+using OWML.ModHelper.Events;
+using OWML.Common;
+using UnityEngine;
+using UnityEngine.UI;
+using System;
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModComboInput : ModInput<string>, IModComboInput
+    {
+        public IModLayoutButton Button { get; }
+        protected readonly IModInputMenu InputMenu;
+        protected readonly TwoButtonToggleElement ToggleElement;
+
+        private string _value;
+        private Vector3 _scale;
+        private UIStyleManager _styleManager;
+        private HorizontalLayoutGroup _layoutGroup;
+        private readonly static int fontSize = 36;
+        private readonly static Vector2 normalPivot = new Vector2(0.5f, 0.5f);
+        private readonly static float scaleDown = 0.75f;
+
+        public ModComboInput(TwoButtonToggleElement element, IModMenu menu, IModInputMenu inputMenu) : base(element, menu)
+        {
+            ToggleElement = element;
+            InputMenu = inputMenu;
+            _scale = element.GetValue<Button>("_buttonTrue").transform.localScale;
+            Button = new ModLayoutButton(element.GetValue<Button>("_buttonTrue"), menu);
+            Button.OnClick += Open;
+            var noButton = ToggleElement.GetValue<Button>("_buttonFalse");
+            noButton.transform.parent.gameObject.SetActive(false);
+            _layoutGroup = Button.LayoutGroup;
+            ((RectTransform)_layoutGroup.transform).sizeDelta = new Vector2(((RectTransform)Button.Button.transform.parent).sizeDelta.x * 2, ((RectTransform)Button.Button.transform.parent).sizeDelta.y);
+            _styleManager = MonoBehaviour.FindObjectOfType<UIStyleManager>();
+
+            var layoutGroup = Button.Button.transform.parent.parent.GetComponent<HorizontalLayoutGroup>();
+            layoutGroup.childControlWidth = true;
+            layoutGroup.childForceExpandWidth = true;
+            Button.Button.transform.parent.GetComponent<LayoutElement>().preferredWidth = 100;
+        }
+
+        private void UpdateLayout(string currentCombination)
+        {
+            int childCount = _layoutGroup.transform.childCount;
+            for (int i = childCount - 1; i >= 0; i--)
+            {
+                GameObject.Destroy(_layoutGroup.transform.GetChild(i).gameObject);
+            }
+            string[] individualCombos = currentCombination.Split('/');
+            for (int i = 0; i < individualCombos.Length; i++)
+            {
+                string[] keyStrings = individualCombos[i].Split('+');
+                for (int j = 0; j < keyStrings.Length; j++)
+                {
+                    Texture2D keyTexture;
+                    if (keyStrings[j].Contains("Xbox_"))
+                    {
+                        keyTexture = InputTranslator.GetButtonTexture((XboxButton)Enum.Parse(typeof(XboxButton), keyStrings[j].Substring(5)));
```

Comment:
I'm guessing the 5 here is the length of `"Xbox_"`? If so, it's better to do something like `var xboxPrefix = "Xbox_"` and then use `xboxPrefix.Length` instead of `5`.

Reply:
done